### ISSUE TITLE
Vault notes can't be accessed despite being indexed, Gemma3 12B via LMStudio

### ISSUE THREAD

**Issue by alliemnguyen** [logancyang/obsidian-copilot#1484]
- [x] Disable all other plugins besides Copilot **(required)**
- [x] Screenshot of note + Copilot chat pane + dev console added **(required)**

Copilot version: v2.8.9

(Bug report without the above will be closed)

_**Describe how to reproduce**_
_A clear and concise description of what the bug is. Clear steps to reproduce the behavior_
1. Install fresh Copilot plugin
2. Add custom model (gemma-3-12b-it from LM Studio)
3. Add custom embedding model (text-embedding-nomic-embed-text-v1.5@q4_k_m from LM Studio)
4. Restart Obsidian
5. Open chat, switch to vault QA, ask question.

**Expected behavior**
_A clear and concise description of what you expected to happen._
Summary of concept from different notes, e.g. here, I have notes on viruses, viral genome, virus quantification, and virus structure. I expect the plugin to pull information from all of these. 1 of these notes is currently opened, but all notes are on the same vault and some are backlinked together too.

**Screenshots**
_Add screenshots to help explain your problem. Please turn on debug mode in Copilot settings, turn off other plugins to leave only Copilot dev messages as necessary._

<img width="766" alt="Image" src="https://github.com/user-attachments/assets/6aadfb50-1a57-4bb9-8e3e-e575e9249833" />

<img width="817" alt="Image" src="https://github.com/user-attachments/assets/d62be201-86e2-4e53-8658-d6f2bc6ec6b1" />

<img width="267" alt="Image" src="https://github.com/user-attachments/assets/05c8c878-0529-4ffd-acf4-ccad6a6bc566" />

**Additional context**
_Add any other context about the problem here._

I can't get Obsidian Copilot to pull information from my notes, even though I'm using vault QA and not the chatbox. It keeps saying there's no "Chat History", or in some other occasions, it says the plugin can't access my notes. I've tried to force vault reindexing, restarting Obsidian, uninstall and reinstall Copilot plugin, but none works. The developer logs in LM Studio doesn't show any warnings. I've verified my models before adding them to the plugin.

System information:
- My model is Gemma 3 12B, installed in LM Studio. 
- My embedding model is text-embedding-nomic-embed-text-v1.5@q4_k_m, also installed in LM Studio
- My laptop is Macbook Pro, M2 Max 32 GB.

The Relevant Notes aren't working either, which I think is also because the plugin can't access my notes.

**Comment by logancyang** [comment]
Thanks for reporting. Please include dev console messages along with your chat panel in your screenshot. Also check the "list indexed files" command's output.

**Comment by alliemnguyen** [comment]
Hi Logan, I've included the screenshot here. Sorry for not including this in the first place, I'm not very tech-savvy so I didn't know where to find this. Is this what you're asking? If not, please let me know what I can provide!
Also 1 note: None of the 5 indexed files the chatbot listed are in my vault. I clicked on each of them and no files were opened.

<img width="1270" alt="Image" src="https://github.com/user-attachments/assets/d18900f1-4878-41b2-8f03-82c42eeba6fd" />

**Comment by logancyang** [comment]
@alliemnguyen thanks for providing the screenshot! No worries this dev console a bit confusing to everyone at the beginning. 

Could you turn on "debug" toggle in Copilot settings under Advanced tab, and share the dev console again? The messages I'm looking for are retrieval related, the yellow ones are not relevant. 

That said, could you try a non-local model first. Small local models are known to hallucinate in vault QA, I don't recommend any local model under 30B of size since we cannot trust its output.

**Comment by alliemnguyen** [comment]
Ah, I see. I tried the command list indexed files and got the following MD file, it looks about right regarding notes in my vault. 

Debug mode was turned on for the following screenshot. The model I tried here was GPT 4.1 with my OpenAI API key. It seems like the plugin still isn't working? The relevant notes section is still red, and the chatbot still says no relevant information can be found.

<img width="1355" alt="Image" src="https://github.com/user-attachments/assets/97224558-cdd3-44aa-941f-ff5c69a01055" />

[Copilot-Indexed-Files-5-9-2025.md](https://github.com/user-attachments/files/20137373/Copilot-Indexed-Files-5-9-2025.md)

**Comment by logancyang** [comment]
@alliemnguyen I see "no chunks found for query...", this means your embedding model is not working. Could you try the default openai embedding model. 

To make local models work one must have a bit of familiarity with this tool first and have an idea of how these components work together. It's hard for others to debug a local setup for you. I recommend sticking with default models as a beginner and then dive deeper into local models once you know how to debug this.

**Comment by logancyang** [comment]
@ihortarasenko Ollama is working fine for me in vault QA. 

<img width="1141" alt="Image" src="https://github.com/user-attachments/assets/699a447c-4bf4-419f-b45d-fbf352f914a4" />

<img width="1117" alt="Image" src="https://github.com/user-attachments/assets/d13bde5c-2517-4769-9f5c-5e646fb7ddb7" />

**Comment by logancyang** [comment]
@ihortarasenko Glad you made it work! No worries. Ollama needs the Ollama client and not the OpenAI format client, they work differently.

**Comment by BigShubham897** [comment]
I am also facing basically the same issue as @alliemnguyen. I am on v2.8.9 of copilot.

I'm using the Obsidian Copilot plugin to interact with a local Large Language Model, Qwen3-8B, and an embedding model, Snowflake/snowflake-arctic-embed-m-v1.5. Both models are running locally through LM Studio. The Qwen3-8B model works well for general chat and can access content when I specifically mention notes in my prompts. However, when I use Copilot's "Vault QA" feature or ask a broad question that requires it to search my entire vault for context, the Qwen3-8B model responds that it "does not have access to my Obsidian vault or its contents."

I've repeatedly attempted to configure the embedding model within Copilot's settings, trying both the API identifier text-embedding-snowflake-arctic-embed-m-v1.5 and the full model name Snowflake/snowflake-arctic-embed-m-v1.5 in the "Model Name" field, but without success. For the "Base URL," I've consistently used 'http://127.0.0.1:1234/v1'

I also did "list indexed files" Copilot command which does indeed show all the files in my vault. Sorry if I didn't describe the issue properly, very new to learning this stuff.

**Comment by ukoehler-sympatec** [comment]
I can report the same issue here as reported by @BigShubham897 and @alliemnguyen in my local installation. I am using lmstudio-community/text-embedding-granite-embedding-278m-multilingual locally on LMStudio. It has worked before but I changed too many parameters simultaneously and have not used is for some time before I noticed that it does not show relevant notes anymore.

- The LMStudio log reports returning vectors when indexing a note.
- List indexed files returned all files in my vault. (I reindexed my vault several times)
- Reset and setup my copilot settings several times
- No suspicious entries in the dev console messages are shown.
- Relevant notes are still not shown.

Finally I reverted back to v2.8.7 and everything works as expected immediately. I can only guess that problems started after updating to v2.8.9 of copilot.

**Comment by BigShubham897** [comment]
I reverted to v2.8.7 based on @ukoehler-sympatec comment. Everything works now.

**Comment by oerteln** [comment]
**Same issue here.** 
I'm using version 2.9.0.
My configuration is text-embedding-nomic-embed-text-v1.5-embedding for the embedding and google/gemma-3-12b for the chat, served via LM Studio.

Indexing is apparently working - I'm seeing log-entries on the Obsidian debug console and requests to the embedding model being made.

When asking a question in 'vault QA' things are getting strange. From the point of view of LM Studio the following happens:

**My theory so far:**

- First interaction: The first request is wrapping my prompt in the chat, but the overall prompt seems to be truncated. Note the 'Standalone question:' at the very end of the string. It seems that Copilot tries to ask the model something about my conversation, but didn't actually pose the question...
- Second interaction: Copilot asks to write a passage about my question (which was 'where have I been on vacation last year?'). The resulting story is a clear hallucination...  This interaction in combination with the next one is actually scary!
- Third interaction: Copilot asks for an embedding of the story that was generated in the second interaction.
- Fourth and last interaction: Copilot asks for an answer to the question I actually posed in the first place.

I assume the message flow was supposed to be something like:
Copilot -> Model: Ah, hi there, there is some message from some crazy guy, can you respond right away based on your context???
Model -> Copilot: Ahm, nope!
Copilot -> Model: Guessed so... Could you please provide me with the essence of the question for embedding? 
Model - Copilot: Here you go!  (Result: Some essence of the question)
Copilot -> Embedding: embed(essence of the question)
Embedding -> Copilot: Vector
Copilot -> VectorDB: Vector
VectorDB -> Copilot: Here you go...
Copilot -> Model: Could you please answer, I provide you with some additional context from the VectorDB?
Model -> Copilot: Here you go! (Answer based on additional context from vault)

**!!! BUT !!!**

the very first interaction screwed it all. The answer is unexpected due to the missing 'Standalone question'. The request for the actual text body to embed is completely screwed. the vector DB does find something - the text that the model hallucinated in the second interaction (Bali, white beaches. sun shining, some drinks in the shade... nice!)

**So, if you are still with me...**

It CAN be reproduced!!! So the label 'cannot be reproduced' is wrong...
And the bug according to my theory has nothing to do with the embedding. I'd assume it is some bug involving this very first interaction from Copilot to the model, making Copilot forget about asking the one important question, making the model respond with, well, what models do when they are missing input - they are dreaming...

In the following the debug log from LM Studio.

Best,
Norbert

`2025-06-19 21:19:30 [DEBUG] 
Received request: POST to /v1/chat/completions with body  {
  "model": "mistralai/mistral-nemo-instruct-2407",
  "temperature": 0.1,
  "stream": true,
  "stream_options": {
    "include_usage": true
  },
  "max_tokens": 19800,
  "messages": [
    {
      "role": "user",
      "content": "Given the following conversation and a follow up q... <Truncated in logs> ...letztes jahr im urlaub?\n\n\n    Standalone question:"
    }
  ]
}
2025-06-19 21:19:30  [INFO] 
[LM STUDIO SERVER] Running chat completion on conversation with 1 messages.
2025-06-19 21:19:30  [INFO] 
[LM STUDIO SERVER] Streaming response...
2025-06-19 21:19:30 [DEBUG] 
Sampling params:	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.100
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
2025-06-19 21:19:30 [DEBUG] 
sampling: 
logits -> logit-bias -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
generate: n_ctx = 6016, n_batch = 512, n_predict = 19800, n_keep = 259
2025-06-19 21:19:30 [DEBUG] 
Detected 2/456 cached tokens to be a valid prefix of the current prompt. Removing 454 tokens from the end of the cache
BeginProcessingPrompt
2025-06-19 21:19:31 [DEBUG] 
FinishedProcessingPrompt. Progress: 100
2025-06-19 21:19:31  [INFO] 
[LM STUDIO SERVER] First token generated. Continuing to stream response..
2025-06-19 21:19:31 [DEBUG] 
target model llama_perf stats:
llama_perf_context_print:        load time =   13081.35 ms
llama_perf_context_print: prompt eval time =     744.81 ms /   257 tokens (    2.90 ms per token,   345.05 tokens per second)
llama_perf_context_print:        eval time =     289.87 ms /    10 runs   (   28.99 ms per token,    34.50 tokens per second)
llama_perf_context_print:       total time =    1038.38 ms /   267 tokens
2025-06-19 21:19:31  [INFO] 
Finished streaming response
2025-06-19 21:19:31 [DEBUG] 
Received request: POST to /v1/chat/completions with body  {
  "model": "mistralai/mistral-nemo-instruct-2407",
  "temperature": 0.1,
  "stream": true,
  "stream_options": {
    "include_usage": true
  },
  "max_tokens": 19800,
  "messages": [
    {
      "role": "user",
      "content": "Human: Please write a passage to answer the questi... <Truncated in logs> ...as the location of my vacation last year?\nPassage:"
    }
  ]
}
2025-06-19 21:19:31  [INFO] 
[LM STUDIO SERVER] Running chat completion on conversation with 1 messages.
2025-06-19 21:19:31  [INFO] 
[LM STUDIO SERVER] Streaming response...
2025-06-19 21:19:31 [DEBUG] 
Sampling params:	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.100
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
2025-06-19 21:19:31 [DEBUG] 
sampling: 
logits -> logit-bias -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
generate: n_ctx = 6016, n_batch = 512, n_predict = 19800, n_keep = 45
2025-06-19 21:19:31 [DEBUG] 
Detected 2/269 cached tokens to be a valid prefix of the current prompt. Removing 267 tokens from the end of the cache
BeginProcessingPrompt
2025-06-19 21:19:31 [DEBUG] 
FinishedProcessingPrompt. Progress: 100
2025-06-19 21:19:31  [INFO] 
[LM STUDIO SERVER] First token generated. Continuing to stream response..
2025-06-19 21:19:33 [DEBUG] 
target model llama_perf stats:
llama_perf_context_print:        load time =   13081.35 ms
llama_perf_context_print: prompt eval time =     138.13 ms /    43 tokens (    3.21 ms per token,   311.30 tokens per second)
llama_perf_context_print:        eval time =    1541.50 ms /    54 runs   (   28.55 ms per token,    35.03 tokens per second)
llama_perf_context_print:       total time =    1694.43 ms /    97 tokens
2025-06-19 21:19:33  [INFO] 
Finished streaming response
2025-06-19 21:19:33 [DEBUG] 
Received request: POST to /v1/embeddings with body  {
  "model": "text-embedding-nomic-embed-text-v1.5-embedding",
  "input": "Last year, your vacation was spent in the vibrant ... <Truncated in logs> ...üell, and enjoyed tapas by the bustling La Rambla.",
  "encoding_format": "base64"
}
2025-06-19 21:19:33  [INFO] 
Received request to embed: Last year, your vacation was spent in the vibrant city of Barcelona, Spain. You explored its charmin...
2025-06-19 21:19:33  [INFO] 
Returning embeddings (not shown in logs)
2025-06-19 21:19:33 [DEBUG] 
Received request: POST to /v1/chat/completions with body  {
  "model": "mistralai/mistral-nemo-instruct-2407",
  "temperature": 0.1,
  "stream": true,
  "stream_options": {
    "include_usage": true
  },
  "max_tokens": 19800,
  "messages": [
    {
      "role": "user",
      "content": "You are Obsidian Copilot, a helpful assistant that... <Truncated in logs> ...: What was the location of my vacation last year?\n"
    }
  ]
}
2025-06-19 21:19:33  [INFO] 
[LM STUDIO SERVER] Running chat completion on conversation with 1 messages.
2025-06-19 21:19:33  [INFO] 
[LM STUDIO SERVER] Streaming response...
2025-06-19 21:19:33 [DEBUG] 
Sampling params:	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.100
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
2025-06-19 21:19:33 [DEBUG] 
sampling: 
logits -> logit-bias -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
generate: n_ctx = 6016, n_batch = 512, n_predict = 19800, n_keep = 436
2025-06-19 21:19:33 [DEBUG] 
Detected 2/99 cached tokens to be a valid prefix of the current prompt. Removing 97 tokens from the end of the cache
2025-06-19 21:19:33 [DEBUG] 
BeginProcessingPrompt
2025-06-19 21:19:34 [DEBUG] 
FinishedProcessingPrompt. Progress: 100
2025-06-19 21:19:34  [INFO] 
[LM STUDIO SERVER] First token generated. Continuing to stream response..
2025-06-19 21:19:34 [DEBUG] 
target model llama_perf stats:
llama_perf_context_print:        load time =   13081.35 ms
llama_perf_context_print: prompt eval time =     806.24 ms /   434 tokens (    1.86 ms per token,   538.30 tokens per second)
llama_perf_context_print:        eval time =     295.04 ms /    10 runs   (   29.50 ms per token,    33.89 tokens per second)
llama_perf_context_print:       total time =    1105.23 ms /   444 tokens
2025-06-19 21:19:34  [INFO] 
Finished streaming response`

**Comment by MikronT** [comment]
Can confirm the issue when using **any** embedding model via LM Studio. Here I'm using `nomic-ai/nomic-embed-text-v1.5`. Looking at the created index it seems like all the generated embeddings are just 0s:

<img width="954" height="131" alt="Image" src="https://github.com/user-attachments/assets/c8ada7fb-8f98-42c7-aee4-cce984b171e7" />

When using the same model over Ollama, the embeddings seem to be stored correctly:

<img width="931" height="349" alt="Image" src="https://github.com/user-attachments/assets/992b2f21-dd92-47ad-b4d6-6b58850cd3a2" />

The LM Studio itself does generate proper embeddings when called via Python or something

<img width="893" height="259" alt="Image" src="https://github.com/user-attachments/assets/375e7f36-02e9-4536-8952-1e2dc6a00496" />

So it's probably something in-between LM Studio and the plugin
